### PR TITLE
Maximum product array

### PR DESCRIPTION
**Pull request by Kooku**
https://leetcode.com/problems/maximum-product-subarray/

### CODE REVIEW

File: python/maximum_product_subarray.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        Brute force solution: O(n^2) Nested for loops trying all possible combinations
+
+        Optimal solution: O(n)
+        Keep a running minimum, maximum product and the max throughout the iteration.
+
+        For the cases where we would have a new max from a positive * positive, we have cur_max
+        For the cases where we would have a new max from a negative * negative, we have cur_min
+        Once we have a new max, we track the value by using total_max and return the total_max in the end.
+        """ 
+        cur_max, cur_min, total_max = nums[0], nums[0], nums[0]
```
Comment (kujinshin) [inline python/maximum_product_subarray.py]: You could do
`cur_max = cur_min = total_max = nums[0]`

File: python/maximum_product_subarray.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+        for index in range(1, len(nums)):
+            num = nums[index]
```
Comment (kujinshin) [inline python/maximum_product_subarray.py]: You could do:
`for num in nums[1:]:`

File: python/maximum_product_subarray.py (hunk 1 above)
Comment (Kooku) [reply python/maximum_product_subarray.py]: I thought by doing that, changing `cur_min` would change `cur_max` which is what happens in Java I believe. This is good to know

File: python/maximum_product_subarray.py (hunk 2 above)
Comment (Kooku) [reply python/maximum_product_subarray.py]: tyty

File: python/maximum_product_subarray.py (hunk 1 above)
Comment (kujinshin) [reply python/maximum_product_subarray.py]: Ahh looks like Java shares the reference(pointer) if you do it this way.

**Pull request merged**